curly.go

```diff
@@ -18,7 +19,7 @@ type CurlyRouter struct{}
 // for the WebService and its Route for the given Request.
 func (c CurlyRouter) SelectRoute(
 	webServices []*WebService,
-	httpRequest *http.Request) (selectedService *WebService, selected *Route, err error) {
+	httpRequest *http.Request) (selectedService *WebService, selected *Route, params map[string]string, err error) {
```

Comment:
this is a breaking change as the exposed function signature is changed.  was this really needed?  I will continue reviewing.

Reply:
Ah, I hadn't appreciated that other people might supply their own router so I thought as long as I updated the two in here this would be OK.  I'd done this as I wanted the router to be responsible for doing the calculation of the parameters as it's them that changes how they do routing and handling of the params.  Instead I think it's probably better to do this through an extra interface that the routers can optionally implement, it's not quite as clean but better than introducing a breaking change...